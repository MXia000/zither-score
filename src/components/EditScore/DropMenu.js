import React, { Component } from "react";
import Dot from '@material-ui/icons/FiberManualRecord';
import Line from '@material-ui/icons/Remove';
import DoubleLine from '@material-ui/icons/DragHandle';

class DropMenu extends Component {
    render() {
        return (
            <div className="dropdown-menu">
                
            </div>
        )
    }
}
  
export default DropMenu;