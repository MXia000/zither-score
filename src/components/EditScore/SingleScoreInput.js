import React, { Component } from "react";
import "../../stylesheets/style.css"
import AddIcon from '@material-ui/icons/Add';
import ControlPointIcon from "@material-ui/icons/ControlPoint";
import MoreHorizIcon from '@material-ui/icons/MoreHoriz';
import Dot from '@material-ui/icons/FiberManualRecord';
import Line from '@material-ui/icons/Remove';
import DoubleLine from '@material-ui/icons/DragHandle';



class SingleScoreInput extends Component {
  render() {
    return (
      <div>
        <div className="dropdown">
          <button
            key="0"
            type="button"
            className="button-style dropdown-toggle"
            data-toggle="dropdown"
            aria-haspopup="true"
            aria-expanded="false"
          >
            <AddIcon fontsize="small"/>
          </button>
            <div className="dropdown-menu">
              <a className="dropdown-item"><Dot /></a>
              <a className="dropdown-item">Double Dot</a>
            </div>
          <button
            key="1"
            type="button"
            className="button-style"
          >
            <AddIcon fontsize="small"/>
          </button>
          <button
            key="2"
            type="button"
            className="button-style"
          >
            <AddIcon fontsize="small"/>
          </button>
          <button
            key="3"
            type="button"
            className="button-style"
          >
            <AddIcon fontsize="small"/>
          </button>
        </div>
        <div>
          <input
            key="0"
            className="singleNote"
            type="number"
            min="0"
            max="7"
          />
          <input
            key="1"
            className="singleNote"
            type="number"
            min="1"
            max="7"
          />
          <input
            key="2"
            className="singleNote"
            type="number"
            min="1"
            max="7"
          />
          <input
            key="3"
            className="singleNote"
            type="number"
            min="1"
            max="7"
          />
          <span className="lineInBetween">|</span>
          <div>
            <button
              key="0"
              type="button"
              className="button-style"
            >
              <AddIcon fontsize="small"/>
            </button>
            <button
              key="1"
              type="button"
              className="button-style"
            >
              <AddIcon fontsize="small"/>
            </button>
            <button
              key="2"
              type="button"
              className="button-style"
            >
              <AddIcon fontsize="small"/>
            </button>
            <button
              key="3"
              type="button"
              className="button-style"
            >
              <AddIcon fontsize="small"/>
            </button>
          </div>
        </div>
      </div>
    );
  }
}

export default SingleScoreInput;