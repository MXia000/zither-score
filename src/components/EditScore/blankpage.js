import React, { Component } from 'react';
import { Auth } from 'aws-amplify';


class blankpage extends Component {
    render () {
        return (
            <div>
                <Container>
                    
                    
                </Container>
            </div>
        )
    }
}
s
export default blankpage;