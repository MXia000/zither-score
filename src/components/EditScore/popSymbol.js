import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Popper from '@material-ui/core/Popper';
import Dot from "@material-ui/icons/FiberManualRecord";
import Line from "@material-ui/icons/Remove";
import DoubleLine from "@material-ui/icons/DragHandle";

const useStyles = makeStyles(theme => ({
  paper: {
    padding: theme.spacing(1),
    backgroundColor: theme.palette.background.paper,
  },
}));

export default function popSymbol() {
  const classes = useStyles();
  const [anchorEl, setAnchorEl] = React.useState(null);

  const handleClick = event => {
    setAnchorEl(anchorEl ? null : event.currentTarget);
  };

  const open = Boolean(anchorEl);
  const id = open ? 'simple-popper' : undefined;

  return (
    <div>    
      <Popper id={id} open={open} anchorEl={anchorEl}>
        <div className={classes.paper}>The content of the Popper.</div>
      </Popper>
    </div>
  );
}
