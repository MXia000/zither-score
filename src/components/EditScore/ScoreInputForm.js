import React, { Component } from "react";
import { SingleScoreInput } from "./SingleScoreInput"

class ScoreInputForm extends Component {
  render() {
    return (
      <ScoreInputForm />
    );
  }
}

export default ScoreInputForm;