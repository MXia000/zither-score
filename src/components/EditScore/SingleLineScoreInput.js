import React from "react";
import SingleScoreInput from "./SingleScoreInput";

// Map through singleInput for one line score input
const SingleLineScoreInput = props => {
  const nodeLength = [0, 1, 2, 3];

  if(props.score) {
    return (
      <SingleScoreInput 
        lineLength={props.lineLength} 
        nodeLength={nodeLength} 
        score={props.score}
      />
    );
  }
};

export default SingleLineScoreInput;

