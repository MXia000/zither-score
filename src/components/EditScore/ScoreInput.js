import React, { Component } from "react";
import SingleLineScoreInput from "./SingleLineScoreInput";
import Container from "@material-ui/core/Container";
import ControlPointIcon from "@material-ui/icons/ControlPoint";

class ScoreInput extends Component {
  constructor(props) {
    super(props);
    this.state = {
      line: 0,
      lineLength: []
    };

    this.handleNewLine = this.handleNewLine.bind(this);
  }

  handleNewLine = event => {
    event.preventDefault();
    this.setState(preState => ({
      line: this.state.line + 1,
      lineLength: [...preState.lineLength, this.state.line]
    }));
  };

  render() {
    return (
      <Container maxWidth="lg">
        <button onClick={this.handleNewLine}>
          <ControlPointIcon />
        </button>
        <form>
          <SingleLineScoreInput
            key={this.state.line}
            lineLength={this.state.lineLength}
          />
        </form>
      </Container>
    );
  }
}

export default ScoreInput;
