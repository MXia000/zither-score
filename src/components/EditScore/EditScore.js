import React, { Component } from 'react';
import ScoreInput from '../ScoreInput/ScoreInput'
import ToolBar from '../ToolBar/ToolBar';
import '../../stylesheets/scorestylesheet.css';
import ScoreInput from '../ScoreInput/ScoreInput';

class EditScore extends Component {

    render () {
        return (
            <div>
                <ToolBar />
                
                <div id="main-content" className="score-scrollable">
                    <svg width="1150" height="1650" className="score-sheet">
                        <g className="page-contents" transform="scale(1.25, 1.25) translate(32.126, 32.126)">
                            <g className="inside-margin" transform="translate(0,0)">
                                <g className="page-title">
                                    <text font-family="Noto Serif TC" font-style="medium" text-anchor="middle" font-size="40" transform="translate(431.024, 80)">ABC</text>
                                </g>
                            </g>
                            
                            <g className="page-main">
                                <foreignObject x="50" y="140" width="1150" height="1650">
                                    <ScoreInput />
                                </foreignObject>
                            </g>
                        </g>
                    </svg>
                </div>            
            </div>
        )
    }

}

export default EditScore;
