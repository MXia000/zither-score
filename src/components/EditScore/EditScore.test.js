import React from 'react';
import { shallow } from 'enzyme';
import EditScore from './EditScore';

describe("EditScore", () => {
    it('render editscore without crashing', () => {  
        const EditScoreComp = shallow(<div><EditScore /></div>);
        expect(EditScoreComp).toMatchSnapshot();
    });
});