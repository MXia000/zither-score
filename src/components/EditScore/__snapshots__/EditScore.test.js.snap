// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditScore render editscore without crashing 1`] = `ShallowWrapper {}`;
